x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockFactory.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.BytesRefArray;
+import org.elasticsearch.compute.data.Block.MvOrdering;
+
+import java.util.BitSet;
+import java.util.List;
+import java.util.ServiceLoader;
+
+public class BlockFactory {
+
+    private static final BlockFactory NON_BREAKING = BlockFactory.getInstance(
+        new NoopCircuitBreaker("noop-esql-breaker"),
+        BigArrays.NON_RECYCLING_INSTANCE
+    );
+
+    private static final BlockFactory GLOBAL = loadGlobalFactory();
+    // new BlockFactory(new NoopCircuitBreaker("esql_noop_breaker"), BigArrays.NON_RECYCLING_INSTANCE);
+
+    private static BlockFactory loadGlobalFactory() {
+        ServiceLoader<BlockFactoryParameters> loader = ServiceLoader.load(
+            BlockFactoryParameters.class,
+            BlockFactory.class.getClassLoader()
+        );
+        List<ServiceLoader.Provider<BlockFactoryParameters>> impls = loader.stream().toList();
+        if (impls.size() != 1) {
+            throw new AssertionError("expected exactly one impl, but got:" + impls);
+        }
+        BlockFactoryParameters params = impls.get(0).get();
+        return new BlockFactory(params.breaker(), params.bigArrays());
+    }
+
+    private final CircuitBreaker breaker;
+
+    private final BigArrays bigArrays;
+
+    private final boolean checkBreaker = true;
```

Comment:
Does this do anything actually? I think it'll always be true, right? Maybe we just remove it?

Reply:
Removed. we can re-add it at some point if we decide that it could be useful. see [f766b1b](https://github.com/elastic/elasticsearch/pull/99657/commits/f766b1b26c0604887f7e02be98ff234a7d44319c)